fineract-provider/src/main/java/org/apache/fineract/portfolio/client/service/ClientWritePlatformServiceJpaRepositoryImpl.java

```diff
@@ -243,9 +243,15 @@ public CommandProcessingResult createClient(final JsonCommand command) {
             }
 
             Staff staff = null;
-            final Long staffId = command.longValueOfParameterNamed(ClientApiConstants.staffIdParamName);
-            if (staffId != null) {
-                staff = this.staffRepository.findByOfficeHierarchyWithNotFoundDetection(staffId, clientOffice.getHierarchy());
+            if (clientParentGroup != null) {
+            	if (clientParentGroup.getParent() != null) {
+            		staff = clientParentGroup.getParent().getStaff();
```

Comment:
If the group doesn't have any staff associated then whatever staff selected while creating a client will be lost